### PR TITLE
Divyanshu | Test | Version 3 - Get all voucher page - Bugs

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cvfh0d1

### CODE REVIEW

File: apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.ts [hunk 1]
```diff
@@ -108,12 +108,12 @@ export class EmailSendDialogComponent implements OnInit, OnDestroy {
             return;
         }
 
-        if (!this.copyTypes?.value?.length && this.voucherType !== VoucherTypeEnum.purchaseOrder) {
+        if (!this.copyTypes?.value?.length && this.voucherType !== VoucherTypeEnum.purchaseOrder && this.voucherType !== VoucherTypeEnum.purchase) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.ts:111]:
```suggestion
        if (!copyTypes?.value?.length && voucherType !== VoucherTypeEnum.purchaseOrder && voucherType !== VoucherTypeEnum.purchase) {
```

File: apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.ts [hunk 2]
```diff
@@ -108,12 +108,12 @@ export class EmailSendDialogComponent implements OnInit, OnDestroy {
...
             this.toasterService.showSnackBar("error", this.localeData?.select_invoice_copy);
             return;
         }
 
-        if ([VoucherTypeEnum.estimate, VoucherTypeEnum.generateEstimate, VoucherTypeEnum.proforma, VoucherTypeEnum.generateProforma, VoucherTypeEnum.purchaseOrder].includes(this.selectedItem?.type ?? this.voucherType)) {
+        if ([VoucherTypeEnum.estimate, VoucherTypeEnum.generateEstimate, VoucherTypeEnum.proforma, VoucherTypeEnum.generateProforma, VoucherTypeEnum.purchaseOrder, VoucherTypeEnum.purchase].includes(this.selectedItem?.type ?? this.voucherType)) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/email-send-dialog/email-send-dialog.component.ts:116]:
```suggestion
        if ([VoucherTypeEnum.estimate, VoucherTypeEnum.generateEstimate, VoucherTypeEnum.proforma, VoucherTypeEnum.generateProforma, VoucherTypeEnum.purchaseOrder, VoucherTypeEnum.purchase].includes(selectedItem?.type ?? voucherType)) {
```

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts [hunk 3]
```diff
@@ -359,12 +359,10 @@ export class BulkUpdateComponent implements OnInit, OnDestroy {
                     let dueDate = "";
 
                     if (this.bulkUpdateForm.get("dueDate")?.value) {
-                        let dueDate = (typeof this.bulkUpdateForm.get("dueDate")?.value === "object") ? dayjs(this.bulkUpdateForm.get("dueDate")?.value).format(GIDDH_DATE_FORMAT) : dayjs(this.bulkUpdateForm.get("dueDate")?.value, GIDDH_DATE_FORMAT).format(GIDDH_DATE_FORMAT);
-
+                        dueDate = (typeof this.bulkUpdateForm.get("dueDate")?.value === "object") ? dayjs(this.bulkUpdateForm.get("dueDate")?.value).format(GIDDH_DATE_FORMAT) : dayjs(this.bulkUpdateForm.get("dueDate")?.value).format(GIDDH_DATE_FORMAT);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts]:
```suggestion
                        dueDate = dayjs(this.bulkUpdateForm.get("dueDate")?.value).format(GIDDH_DATE_FORMAT);
```

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 4]
```diff
@@ -473,6 +479,7 @@ export class VoucherListComponent implements OnInit, OnDestroy {
 
         this.componentStore.sendEmailIsSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
             if (response) {
+                this.sendEmailModalDialogRef.close();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts]:
```suggestion
                this.sendEmailModalDialogRef?.close();
```

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 5]
```diff
@@ -577,7 +596,7 @@ export class VoucherListComponent implements OnInit, OnDestroy {
      * @memberof VoucherListComponent
      */
     private handleGetAllVoucherResponse(response: any): void {
-        if (response) {
+        if (response && response?.voucherType === this.voucherType) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts:599]:
```suggestion
        if (response && response.voucherType === this.voucherType) {
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/bulk-update/bulk-update.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/list/list.component.ts (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/list/list.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/list/list.component.ts (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/list/list.component.ts:599]: No need here

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**